files/qsub.pl

```diff
@@ -428,25 +428,30 @@ sub make_command
     # TODO: support all/half for both -l gpus and --gpus
     if ($res_opts->{naccelerators}) {
         if ($gpus) {
-            fatal("You cannot define both :gpu=X node resource and the --gpus option")
+            fatal("You cannot define both :gpus=X/-l gpus=X node resource and the --gpus option")
         } else {
             push(@command, "--gres=gpu:$res_opts->{naccelerators}");
+            push(@intcommand, '--gres=gpu:0') if $interactive;
         }
+
+        if ($cpus_per_gpu) {
+            fatal("--cpus-per-gpu requires --gpus option (and thus conflicts with :gpus/-l gpus")
+        }
+
     } elsif ($gpus) {
         push(@command, "--gpus=$gpus");
+        push(@intcommand, '--gpus=0') if $interactive;
+        # apparently, only when --gpus is set (according to man page)
+        push(@command, "--cpus-per-gpu=$cpus_per_gpu") if $cpus_per_gpu;
+        # no equivalent for interactive?
     }
 
     # TODO: handle node resource GPUs too. might/will conflict with node resourfces such as vmem etc etc
     #       needs support in submitfilter too?
     #       This is a way too simple first attempt. needs to be understood how slurm handles these combos
-    # TODO: joltik defaults for now, to be read from vsc-jobs clusterdata
-    $cpus_per_gpu = 8 if !$cpus_per_gpu;
-    push(@command, "--cpus-per-gpu=$cpus_per_gpu") if
-        $gpus || ($res_opts->{naccelerators} && ! $res_opts->{mppnppn});
-
-    $mem_per_gpu = "48G" if !$mem_per_gpu;
-    push(@command, "--mem-per-gpu=".convert_mb_format($mem_per_gpu)) if
-        $gpus || ($res_opts->{naccelerators} && ! ($res_opts->{mem} || $res_opts->{pmem}));
+    my $add_mempergpu = $gpus || ($res_opts->{naccelerators} && ! ($res_opts->{mem} || $res_opts->{pmem}));
+    push(@command, "--mem-per-gpu=".convert_mb_format($mem_per_gpu)) if $mem_per_gpu && $add_mempergpu;
+    push(@intcommand, "--mem-per-gpu=0") if $interactive && $add_mempergpu;  # TODO: always add this when gpus or nacc is defined?
```

Comment:
Yes, I think you should always add this.

See also fix in https://github.com/hpcugent/slurm-torque-wrappers/pull/4

Reply:
done